pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -74,16 +76,98 @@ type cloudWatchExecutor struct {
 
 func (e *cloudWatchExecutor) newSession(region string) (*session.Session, error) {
 	dsInfo := e.getDSInfo(region)
-	creds, err := getCredentials(dsInfo)
+
+	cacheKey := fmt.Sprintf("%d:%s:%s:%s:%s:%s", dsInfo.AuthType, dsInfo.AccessKey, dsInfo.Profile, dsInfo.AssumeRoleARN,
+		dsInfo.ExternalID, region)
+
+	sessCacheLock.RLock()
+	if env, ok := sessCache[cacheKey]; ok {
+		if env.expiration.After(time.Now().UTC()) {
+			sessCacheLock.RUnlock()
+			plog.Debug("Using cached AWS session", "key", cacheKey)
```

Comment:
@marefr How high frequency are we talking here, once per request? If so, it's not very high for a debug message is it? I think the reason @papagian wanted this is to help exactly debugging, in case a cached session is used and it shouldn't be (I think she was experiencing this).

@papagian Comments?

Reply:
I think that would be helpful for debugging (and actually it was helpful when I tested the PR as well). It's a debug message so I expect that most users not to have it on and only enable it whenever they have an issue they want to debug.  I understand though that for the rest of users can be just noise.
Personally I would like it to stay but I don't insist.

Another concern I had related to this message is whether any of the logged information is considered sensitive
and it has to be redacted.

Reply:
@papagian I think the only potentially sensitive part of the key is the access key ID, but isn't that public? It's the secret part of the key pair that's, well, secret.

Reply:
Maybe should wrap the log statement in `if setting.Env == setting.Dev { <log> }` if these logs are for local debug purposes? I know that many users and customers are always running with full debug enabled that's why I'm careful.

Reply:
I don't think it's only for local logging, but I'll leave it to @papagian  to answer since she asked for it. 

If it's only as frequent as at most every request (as I suspect), why would having a debug message be a problem? It's not like we don't have debug logging already, we need them for diagnostic purposes. If we can't add debug messages that aren't too chatty, we're going to make troubleshooting very difficult or ourselves.

Reply:
My initiative for this was if a user reports authentication failures to be able to tell easily from the logs if the cache was used. I don't think that users can/should enable easily dev environment. We can leave it for now and adding later if we have hard time debugging such cases.

Reply:
@papagian OK by me. The question is now if I shall remove the debug log statement or not.

Reply:
> @papagian OK by me. The question is now if I shall remove the debug log statement or not. To me it's the same.

let's remove it for now